core/src/components/picker-column/picker-column.tsx

```diff
@@ -70,7 +73,7 @@ export class PickerColumn implements ComponentInterface {
   /**
    * Emitted when the value has changed.
    */
-  @Event() ionChange!: EventEmitter<PickerColumnItem>;
+  @Event() ionChange!: EventEmitter<{ value: string | number | undefined }>;
```

Comment:
Sorry, I had commented regarding this in https://github.com/ionic-team/ionic-framework/pull/28622#discussion_r1412601566, but it was marked as outdated.

TL;DR; This will likely change again in order to address the keyboard entry behavior, so I'd prefer to keep this as-is for now. Whatever type we end up converging on I'll make sure the implementations of it are cleaned up.

Reply:
Sounds good, I'm ok with addressing this later 👍